src/librustdoc/html/render/print_item.rs

```diff
@@ -417,7 +411,7 @@ fn item_module(w: &mut Buffer, cx: &Context<'_>, item: &clean::Item, items: &[cl
         }
     }
 
-    if curty.is_some() {
+    if last_section.is_some() {
         w.write_str(ITEM_TABLE_CLOSE);
     }
```

Comment:
There are some slight diffs to the conditions under which ITEM_TABLE_CLOSE gets emitted. It seems like probably we are emitting too many closing tags. Under this theory, the browser is trying to fix up the HTML structure as best it can, and does surprising well... but winds up with the struct links not nested under `<section id="main-content" class="content">`. Since the colors are only applied to children of `.content`, they are no longer applied here.

Reply:
Ah, that'd explain the HTML parse errors in the tests then.

Reply:
It ended up being a very simple and (in retrospect) obvious fix :)

I think I had gotten confused by the old version of the code since it used `myty` and `curty`, which sound like they're the same but aren't.

```diff
diff --git a/src/librustdoc/html/render/print_item.rs b/src/librustdoc/html/render/print_item.rs
index 86d070a99a1..864035f2323 100644
--- a/src/librustdoc/html/render/print_item.rs
+++ b/src/librustdoc/html/render/print_item.rs
@@ -286,7 +286,9 @@ fn cmp(

         let my_section = item_ty_to_section(myitem.type_());
         if Some(my_section) != last_section {
-            w.write_str(ITEM_TABLE_CLOSE);
+            if last_section.is_some() {
+                w.write_str(ITEM_TABLE_CLOSE);
+            }
             last_section = Some(my_section);
             write!(
                 w,
```